blockchain/state_migration.go

```diff
@@ -371,3 +372,132 @@ func (bc *BlockChain) StopStateMigration() error {
 func (bc *BlockChain) StatusStateMigration() (bool, uint64, int, int, float64) {
 	return bc.db.InMigration(), bc.db.MigrationBlockNumber(), bc.committedCnt, bc.pendingCnt, bc.progress
 }
+
+func (bc *BlockChain) concurrentIterateTrie(root common.Hash, db state.Database, resultCh chan common.Hash, finishCh chan error) (resultErr error) {
+	defer func() {
+		finishCh <- resultErr
+	}()
+
+	stateDB, err := state.New(root, db)
+	if err != nil {
+		return err
+	}
+
+	it := state.NewNodeIterator(stateDB)
+	for it.Next() {
+		resultCh <- it.Hash
+
+		select {
+		case <-bc.quitWarmUp:
+			return errors.New("quitWarmUp")
+		case <-bc.quit:
```

Comment:
If `StopWarmUp()` is called by `BlockChain.Stop`, this case won't be needed.

Reply:
Right. It can be written as you mentioned. 
I think it is a another option. :)  
The two ways don't seem to make a big difference.